applications/server/modules/access.py

```diff
@@ -0,0 +1,83 @@
+from gluon import current
+from gluon import auth
+
+
+# Permission types.
+# v = view
+# a = admin (valid only for one whole device)
+# e = edit settings of procedure
+
+def register_device(name=None, description=''):
+    db = current.db
+    devices = db.devices
+    permission = db.user_permission
+    device_id = devices.insert(name=name, user_id=auth.user_id, description=description)
+    permission.insert(perm_user_email=auth.user.email, device_id=device_id, procedure_id=None, perm_type='a')
+
+
+# Share one device to multiple users with one permission type
+# return emails which are not successfully shared.
+def share_device(device_id, user_email=[], type='v', procedure_id=None):
+    db = current.db
+    permission = db.user_permission
+    fail_email = []
+    for email in user_email:
+        # Does the email exist in registered user table?
+        # if it does not exist this permission adding operation is denied
```

Comment:
This is perhaps a bad idea.  It can be very useful to share a device with a user that has not yet created an account.  I would not have this check.